### PR TITLE
SERVER-24580 Update oldest txn ID with "strict, nowait" flags.

### PR DESCRIPTION
**Pull request by michaelcahill**
The earlier change in SERVER-24580 made the eviction server not force update to the oldest transaction ID.  That was done to avoid a performance collapse with many threads in the library (every run of the eviction server had to wait for all threads to exit the snapshot table and blocked all threads from entering it).

Previously "force" meant both (a) "update no matter how far behind the oldest has fallen" and (b) "wait for locks".  This change splits those apart, calling (a) "strict" and (b) "wait".  It sets strict, no-wait for the eviction server updates to the oldest ID.

### CODE REVIEW

File: src/evict/evict_lru.c [hunk 1]
```diff
@@ -631,7 +631,7 @@ __evict_pass(WT_SESSION_IMPL *session)
 		 * of whether the cache is full, to prevent the oldest ID
 		 * falling too far behind.
 		 */
-		WT_RET(__wt_txn_update_oldest(session, false));
+		WT_RET(__wt_txn_update_oldest(session, true, false));
```
Comment (michaelcahill) [inline src/evict/evict_lru.c]: This is the key fix -- the eviction server should update the oldest transaction ID any time it has fallen behind, but not wait for locks on the snapshot table.

File: src/evict/evict_page.c [hunk 2]
```diff
@@ -420,7 +420,7 @@ __evict_review(
 		 * fallen behind current.
 		 */
 		if (modified)
-			WT_RET(__wt_txn_update_oldest(session, false));
+			WT_RET(__wt_txn_update_oldest(session, true, false));
```
Comment (michaelcahill) [inline src/evict/evict_page.c]: This change is arguable -- before evicting a page, be strict about updating the oldest ID but don't wait for a lock on the snapshot table.  I could go either way on this one based on further perf testing, but I think this change is safe and better than what we were doing before.

**Comment by michaelcahill** [comment]
@agorrod, can you please review?

**Comment by agorrod** [comment]
@michaelcahill the functional changes lgtm. I think it would be more maintainable to use named flags rather than a pair of booleans, but that's personal preference.

**Comment by agorrod** [comment]
Thanks @michaelcahill lgtm - I'll merge.

**Pull request merged**